test/e2e/network/service.go

```diff
@@ -2027,6 +2027,225 @@ var _ = common.SIGDescribe("Services", func() {
+			framework.ExpectNoError(err, "failed to delete service: %s in namespace: %s", serviceName, ns)
+		}()
+		serviceIP := svc.Spec.ClusterIP
+		framework.Logf("internal-traffic-policy-test cluster ip: %s", serviceIP)
+		ginkgo.By("Picking 2 Nodes to test whether internal traffic policy is respected")
+		nodes, err := e2enode.GetBoundedReadySchedulableNodes(cs, 2)
+		framework.ExpectNoError(err)
+		nodeCounts := len(nodes.Items)
+		if nodeCounts < 2 {
+			e2eskipper.Skipf("The test requires at least two ready nodes on %s, but found %v", framework.TestContext.Provider, nodeCounts)
+		}
+
+		ginkgo.By("Creating 1 webserver pods on first node to be part of the TCP service")
+		webserverPod0 := e2epod.NewAgnhostPod(ns, "echo-hostname-0", nil, nil, nil, "netexec", "--http-port", strconv.Itoa(servicePort))
+		webserverPod0.Labels = jig.Labels
+		webserverPod0.Spec.NodeName = nodes.Items[0].Name
+		_, err = cs.CoreV1().Pods(ns).Create(context.TODO(), webserverPod0, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+		framework.ExpectNoError(e2epod.WaitTimeoutForPodReadyInNamespace(f.ClientSet, webserverPod0.Name, f.Namespace.Name, framework.PodStartTimeout))
+
+		defer func() {
+			framework.Logf("Cleaning up the echo server pod")
+			err := cs.CoreV1().Pods(ns).Delete(context.TODO(), webserverPod0.Name, metav1.DeleteOptions{})
+			framework.ExpectNoError(err, "failed to delete pod: %s on node", webserverPod0.Name)
+		}()
+
+		validateEndpointsPortsOrFail(cs, ns, serviceName, portsByPodName{
+			webserverPod0.Name: {servicePort},
+		})
+
+		ginkgo.By("Creating 1 pause pod on second node that will try to connect to the webservers")
+		pausePod0 := &v1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "pause-pod-0",
+				Namespace: ns,
+			},
+			Spec: v1.PodSpec{
+				NodeName: nodes.Items[1].Name,
+				Containers: []v1.Container{
+					e2epod.NewAgnhostContainer("agnhost-pause", nil, nil, "pause"),
+				},
+			},
+		}
+		pausePod0, err = cs.CoreV1().Pods(ns).Create(context.TODO(), pausePod0, metav1.CreateOptions{})
+		framework.ExpectNoError(err)
+		framework.ExpectNoError(e2epod.WaitTimeoutForPodReadyInNamespace(f.ClientSet, pausePod0.Name, f.Namespace.Name, framework.PodStartTimeout))
+
+		defer func() {
+			framework.Logf("Deleting pause pods")
+			err := cs.CoreV1().Pods(ns).Delete(context.TODO(), "pause-pod-0", metav1.DeleteOptions{})
+			framework.ExpectNoError(err, "failed to delete pod: %s on node", "pause-pod")
+		}()
+
+		// Attempt to connect to service cluster IP from pause Pod several times. This should time out every time.
+		serviceAddress := net.JoinHostPort(serviceIP, strconv.Itoa(servicePort))
+		for i := 0; i < 3; i++ {
+			cmd := fmt.Sprintf(`curl -q -s --connect-timeout 30 %s/hostname`, serviceAddress)
+			_, err := framework.RunHostCmd(pausePod0.Namespace, pausePod0.Name, cmd)
+			framework.ExpectError(err, "expected error when trying to connect to cluster IP")
```

Comment:
I don't know if we should conflate both tests, my rationale is that this test will pass even if the pods are not scheduled, however, if we schedule one pod in one node we can:
- test from the same node, it must reply
- test from a different node, it must timeout, right?

Reply:
I think that's a good point. Will combine these two tests